tools/d2g/d2gtest/d2g_test.go

```diff
@@ -241,6 +252,24 @@ func unmarshalYAML(t *testing.T, dest interface{}, path string) {
 	}
 }
 
+func marshalYAML(t *testing.T, src interface{}, path string) {
+	t.Helper()
+	j, err := json.Marshal(src)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	result, err := yaml.JSONToYAML(j)
```

Comment:
This is really nice. Such a shame JSONToYAML doesn't offer options for customising line wrapping etc. But I think it is still worth it, even if it makes the test yamls harder to read.